### DISCUSSION TITLE
About T-matrix and S-matrix, and how to obtain the electromagnetic field distribution of the diffraction structure under the specified amplitude ?

### DISCUSSION THREAD

**Discussion by xungu1** [kc-ml2/meent#78] [Q&A] [answered]
I noticed that your program uses the enhanced T-matrix method to calculate the diffraction efficiency when calculating the multi-layer structure, and I also noticed that your program seems to provide a calculation method for the S-matrix. I currently have two doubts : The first is if I want to get the S-matrix at the interface of each interface of the multi-layer structure, how do I operate ? Can I use the calculation results of T-matrix to get the S-matrix at the interface of each interface ? If so, how should I operate ? The other is how do I operate if I want to get the effect of a specific amplitude on a certain layer of structure ? I have noticed that the S-matrix can be used as a calculation when there are inputs on both sides of the interface. I want to use this S-matrix to calculate the electromagnetic field energy distribution of the layer under different illumination on both sides at the same time, or can the same operation be achieved in another way ?

**Comment by yonghakim** [comment]
SMM option in meent was disabled. I will try to make it available again (or make it sure if it's already available) but cannot say when.

Regarding two questions,
* S matrix
  * SMM and TMM are about how the waves of in and out are connected at the interfaces. S matrix is not defined in TMM. You can utilize other information of Enhanced TMM in meent to calculate S matrix. If you need S matrix, I recommend you TORCWA which is open source python RCWA tool. This uses classical SMM.
* Both sides
  * Meent doesn't support this. If what you mean is to calculate two situations (one situation with incident light from the top and one other situation with incident light from the bottom) with single simulation, not one situation with two light sources, then I suggest to look reticolo(matlab) or TORCWA(python). They are both open source so you can take a look. TORCWA uses classical SMM.

**Reply by xungu1** [reply]
Thank you for your answer. 
For the first question, I will try to calculate the S-matrix by using the correlation of the enhanced T-matrix. 
On the second question, I should be unclear. I want to obtain the electromagnetic field distribution inside the diffraction structure under a given amplitude. This ' given amplitude ' contains both real and imaginary parts. In the previous question, you pointed out that the amplitude of the incident wave in the strictly coupled wave is normalized to 1. Then, in the case of normal incidence, is the electromagnetic field distribution inside the diffraction structure under the ' given amplitude ' multiplied by the current calculated electromagnetic field distribution multiplied by ' the proportion of the given amplitude to the normalized amplitude ' ?

**Reply by yonghakim** [reply]
I'm afraid it cannot be done in such a way. It probably can be done in the code to change the amplitude of the input but am not sure if it is correct in terms of RCWA steps. This is not considered in meent.

Do you have any paper or material about this input amplitude? How did other researchers simulate this?

**Reply by xungu1** [reply]
In the third chapter of ' On design strategies for binary dielectric metasurfaces based on the Fourier modal method ' in the literature I refer to, the S-matrix at each interface of the multilayer structure is calculated to calculate the electromagnetic field distribution inside the diffraction structure, and the transmission amplitude and reflection amplitude under given illumination conditions can be obtained. The proposed S matrix can achieve the effect of simultaneous illumination of two light sources as you mentioned earlier, and it is shown in the text that the S matrix is converted from the T matrix ( although I am still not clear about the details of this conversion ). In the calculation of the S matrix, the weight of the forward propagation and back propagation modes on both sides of the interface is included. 
Is the internal electromagnetic field distribution of the diffraction structure under the so-called ' given amplitude ' calculated from this ? Or if I want to obtain the influence under the ' given amplitude ', I have to rely on the S matrix calculation method when two light sources can be illuminated at the same time ?

**Reply by yonghakim** [reply]
https://opg.optica.org/josaa/abstract.cfm?uri=josaa-34-10-1769

This seems what you are looking for, and doable with reasonable amount of effort. You don't need S-matrix if this can be done with ETM, right?

**Reply by xungu1** [reply]
Sincerely thanks for your help ! I think that the bidirectional incidence mentioned in this paper, and the result of the reverse incidence can be obtained by the matrix calculated by the forward incidence, which is what I am looking for. I have the following questions about RCWA and this paper : In the fourth section of this paper, I have a general understanding of the derivation process, but I do not know the expression of S1 and S2 variables given in Formula ( 14 ) and Formula ( 16 ). Taking TM polarization as an example, Formula ( 8 ) in this paper gives its normalized expression, but it seems that it still needs to be modified when it is substituted in practice. I 'm sorry, is this what I 'm thinking about now, with the formula ( 41 ) in Moharam 's article [ 1 ] as a reference, to reverse the expression of S1 in formula ( 14 ) ? At the same time, the same expression can be set when the TM polarization is also set when the reverse incidence is also set ?
![image](https://github.com/user-attachments/assets/7f4f5c2a-a5e1-4240-82e1-595bf1aee685)
![image](https://github.com/user-attachments/assets/c945b056-a312-4d38-90af-4be0f2322a18)
![image](https://github.com/user-attachments/assets/06dcaab3-a1c1-44ae-9949-978097ef6634)
[1]https://opg.optica.org/josaa/abstract.cfm?uri=josaa-12-5-1068

**Reply by yonghakim** [reply]
>Thank you sincerely for your help! I believe that the bidirectional incidence discussed in this paper, as well as the results for reverse incidence, can be obtained by using the matrix derived from the forward incidence, which is exactly what I’m looking for.
I have a few questions regarding RCWA and the content of this paper. In Section 4, I generally understand the derivation process, but I am unsure about the expressions for the variables $S_1$ and $S_2$ given in Formulas (14) and (16). Using TM polarization as an example, Formula (8) provides the normalized expression, but it seems this might need modification when applied in practice.

>Sorry for the confusion, but am I correct in thinking that Formula (41) from Moharam’s paper [1] can be used to reverse the expression for $S_1$ ​in Formula (14)?

Moharam's (41) is connected to Junker's (14). In (41), there is one light source which has only 0th mode, hence it becomes dirac-delta function. $S_1$ is generalized term of this light source: this is not necessarily 0th mode only. 
For comparison, if $S_I$ is delta function, then (14) is same as (41).

>Additionally, can a similar approach be applied when considering reverse incidence for TM polarization?

I'm not sure what reverse incidence means but I think you can stack the layer in reversed order (upside down) to simulate the reverse incidence.

**Reply by yonghakim** [reply]
$S_2$ seems the light source you are looking for - reverse incidence. It is 0 in Moharam's.

**Reply by xungu1** [reply]
As you say，if SI is delta function, then (14) is same as (41).I think that the reverse incidence can be achieved by flipping the stack, but in fact, whether I pass through the reverse incidence or flip the stack, I cannot bypass the light source that requires the incident light source to be a specified transmission amplitude.What I fail to understand is, if I want to use the specified transmission amplitude as the amplitude of the reverse incident light source, how do I set SI ? According to the data I have consulted, the diffraction efficiency of a certain order of the grating can be defined as the product of the transmission amplitude, that is, | t ^ 2 | = η, where t is the transmission amplitude and η is the diffraction efficiency. So now if I want to set the transmission amplitude to the amplitude of the reverse incident light source, how do I set it.

**Reply by xungu1** [reply]
I have tried to modify the delta function to 2, but the fact is that this is only the result multiplied by 2.
Could you please provide your contact information conveniently ? Such as Facebook / twitter / whatsapp or other contact information, so that we can contact you in a timely manner.

**Reply by yonghakim** [reply, marked as answer]
>| t ^ 2 | = η, where t is the transmission amplitude and η is the diffraction efficiency.

This is not true. See (67) in meent paper. There is more term.

>I have tried to modify the delta function to 2, but the fact is that this is only the result multiplied by 2.

I never tried other source other than delta function so I can't say it's right or wrong.
But it seems reasonable to me that you got a doubled result with a doubled input. What kind of result were you expecting?

**Comment by xungu1** [comment]
Thank you very much for your previous answers ! I simulated aluminum in the visible light band. The real and imaginary parts of the refractive index of aluminum at different wavelengths were used as input. When viewing the transmission and reflection, it was found that the reflectivity would exceed one at some wavelengths. Is this a normal phenomenon ?
Shouldn 't I take the imaginary part as input ? I set the duty cycle of the aluminum grating is 80 %, the period is 350 nm, the wavelength is from 400-750 nm, and the depth is 35 nm.
![result](https://github.com/user-attachments/assets/cc8f854a-9073-4f57-bde3-e53f5e0a2498)

**Reply by yonghakim** [reply]
The sum of the diffraction efficiencies is one when the refractive indices are real.
Whether to use the imaginary part or not, I suggest to follow other precedent works in the same domain.

**Reply by xungu1** [reply]
I think so too.I 'm also going to try some previous examples.Thank you very much！
